### PR TITLE
Add parallel processing & refactor main

### PR DESCRIPTION
**Pull request by Arkay92**
# Summary of Enhancements:

## Error Handling:

- Added try-except blocks to handle potential file loading errors.
- Enhanced error messages for better debugging.

## Parallelization:

- Ensured the use of multiple workers for data loading.
- Confirmed the use of torch.nn.DataParallel for multi-GPU training.

## Optimizations:

- Added constants for batch size and number of workers for easier configuration.
- Streamlined the logging setup process.
- Optimized learning rate adjustment.
- Documentation and Comments:

Added docstrings and comments for better understanding and maintainability.

## General Code Cleanup:

Refactored code for better readability and maintainability.

### CODE REVIEW

**Comment by LivXue** [comment]
Million thanks! I accepted your pull request and fixed some bugs.

**Comment by Arkay92** [comment]
> Million thanks! I accepted your pull request and fixed some bugs.

Anytime, will try see if I can help some more soon !
